src/Auth.php

```diff
@@ -277,23 +277,39 @@ public function connection_ldap($ldap_method, $login, $password)
 
             $dn = $infos['dn'];
             $this->user_found = $dn != '';
-            if ($this->user_found && @ldap_bind($this->ldap_connection, $dn, $password)) {
-               //Hook to implement to restrict access by checking the ldap directory
+
+            if ($this->user_found) {
+                $this->user_found = (bool)@ldap_bind($this->ldap_connection, $dn, $password);
+                if (!$this->user_found) {
+                    $this->addToError(
+                        AuthLDAP::buildError(
+                            $this->ldap_connection,
+                            sprintf(
+                                'Unable to bind %s',
+                                $dn
+                            )
+                        )
+                    );
+                }
+            }
+
+            if ($this->user_found) {
```

Comment:
IMHO, `$this->user_found` should remain to `true` even if `ldap_bind` fails. Indeed, user is still present in LDAP, as we found it, but he probably used a wrong password. Asserting that it is not anymore present in LDAP directory may lead to its deactivation/removal/....

```suggestion
            if ($this->user_found) {
                $bind_result = @ldap_bind($this->ldap_connection, $dn, $password) === false

                if ($bind_result === false) {
                    $this->addToError(
                        AuthLDAP::buildError(
                            $this->ldap_connection,
                            sprintf(
                                'Unable to bind %s',
                                $dn
                            )
                        )
                    );
                }
            }

            if ($this->user_found && $bind_result !== false) {
```